libvast/vast/taxonomies.hpp

```diff
@@ -35,23 +35,22 @@ struct concept_ {
 
   friend bool operator==(const concept_& lhs, const concept_& rhs);
 
+  /// A concept is a Monoid.
+  friend concept_ mappend(concept_ lhs, concept_ rhs);
+
   template <class Inspector>
   friend auto inspect(Inspector& f, concept_& c) {
-    return f(caf::meta::type_name("concept"), c.fields, c.concepts);
+    return f(caf::meta::type_name("concept"), c.description, c.fields,
+             c.concepts);
   }
 };
 
 /// Maps concept names to their definitions.
 using concepts_map = detail::stable_map<std::string, concept_>;
 
-/// Converts a data record to a concept.
-caf::error convert(const data& d, concepts_map& out);
-
-/// Extracts a concept definition from a data object.
-caf::error extract_concepts(const data& d, concepts_map& out);
-
-/// Extracts a concept definition from a data object.
-caf::expected<concepts_map> extract_concepts(const data& d);
+/// Describes the layout of a vast::list of concepts for automatic conversion to
+/// a `concepts_map`.
+extern const list_type concepts_data_layout;
```

Comment:
I'm not convinced of that, you're normally only interested in the layout of the type you're currently working with. In case you want to see them all you can build something with ripgrep:
```sh
rg -IN "extern const [[:alpha:]]+_type (.+_layout);\$" -r '$1' | xargs -I {} rg --multiline -Po '(?s){} = .*?;'
```